monitor/tests/test_monitor.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+# test_monitor.py
+#
+# Copyright (C) 2019 Kano Computing Ltd.
+# License:   http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License v2
+#
+
+import unittest
+import os
+
+class TestMonitor(unittest.TestCase):
```

Comment:
This is great! I would suggest that you take a look at using `pytest` instead of `unittest`. It is basically the same but a little clearer with what it is doing and easier to work with as you can do things like shared fixtures and the like. It also has the benefit that it is what we use in our other projects.

To convert this, you would just have to do the following:
 - Ditch the class and just make the `test_*` just module functions - the file is the grouping element for your tests
 - You don't need to do the `self.assertEqual()` anymore - you just `assert x == y` which makes it super-clear
 - Just run `pytest` from the root directory to run all tests (or `pytest /path/to/file.py` to run an individual test) - no need for any `if __name__ == '__main__':` business

This file would end up looking something like this:

```python
#
# test_monitor.py
#
# Copyright (C) 2019 Kano Computing Ltd.
# License:   http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License v2
#

import os


def _set_command_line(logfile, samples, threshold):
    return 'python ../drfeedback-monitor.py process --logfile={} ' \
        '--samples={} --threshold={}'.format(logfile, samples, threshold)

def test_spike_two_days():
    rc = os.system(_set_command_line('../log_sample_spike.txt', 2, 1.5))
    assert os.WEXITSTATUS(rc) == 1

def test_spike_three_days():
    rc = os.system(_set_command_line('../log_sample_spike.txt', 3, 1.5))
     assert os.WEXITSTATUS(rc) == 1

def test_spike_five_days():
    rc = os.system(_set_command_line('../log_sample_spike.txt', 5, 1.1))
    assert os.WEXITSTATUS(rc) == 1

def test_steady_two_days():
    rc = os.system(_set_command_line('../log_sample_steady.txt', 2, 1.5))
    assert os.WEXITSTATUS(rc) == 0

def test_steady_three_days():
    rc = os.system(_set_command_line('../log_sample_steady.txt', 3, 1.5))
    assert os.WEXITSTATUS(rc) == 0

def test_steady_five_days():
    rc = os.system(_set_command_line('../log_sample_steady.txt', 5, 1.5))
    assert os.WEXITSTATUS(rc) == 0

def test_alert():
    alert_message = 'Slack alert signal requested, simulate=True'
    command_line = _set_command_line('../log_sample_spike.txt', 5, 1.1)
    command_line += ' --verbose'

    output = os.popen(command_line).read()
    assert output.find(alert_message) != -1
    assert output.find('curl') != -1

```

And then `pytest` to run (`pip install pytest` to get the tool installed).

Reply:
This is a lot cleaner! migrated - thanks!